### DISCUSSION TITLE
The experimental ones have been for a long time

### DISCUSSION THREAD

**Discussion by BridgeHeadland** [joncampbell123/dosbox-x#5280] [General]
It seems to me that the experimental and unfinished sections and settings have stayed like this for a long time, so I wonder: have they been worked on from time to time, or have they been put on hold?
I'm specifically referring to ACPI, Gravis Ultrasound Max/Interwave, IDE controller chips, S3 ViRGE/ViRGE VX, and ATI in general.
If I'm not mistaken, I think DOSBox-X needs to be able to emulate the Windows Sound System for the GUS Max emulation to be complete, and possibly GUS Interwave to be slightly better, but not completely complete.

**Comment by joncampbell123** [comment]
The ACPI implementation is very experimental. It was there to make sure I could encode AML properly in the first place by placing AML into the BIOS area and then using a DOS tool to extract it and examine it on Linux with acpicatools.

It is by far not complete enough to function as a proper ACPI BIOS.

**Comment by joncampbell123** [comment]
Gravis Ultrasound Interwave/MAX emulation requires emulating support for >1MB of RAM, the lack of sample rate change based on number of voices (unless in classic GUS emulation), and yes, it's going to need a WSS codec implementation.

And if there is WSS codec implementation, the SC400 Sound Blaster emulation mode would also use it because those cards were basically a CS4231 that provides both Pro and WSS I/O ports. SC400 emulation is based on the sound card I used to have back in the mid 1990s (and still have today, in fact), which was very Sound Blaster Pro compatible, but also had Windows 3.1 WSS drivers, an integrated CD-ROM drive interface, and it needed a DOS program and/or device driver to program the Sound Blaster and WSS resources as well (only the base SB I/O port was set by jumper), this card having been made before ISA Plug & Play was standardized.

**Comment by joncampbell123** [comment]
ATI emulation. I attempted to work on ATI emulation combined with using an ATI VGA Wonder ROM BIOS image from the 86box/PCem ROM collection Git repo here on Github. One of the things the BIOS does is read configuration I/O ports which I implemented, but got stuck on a problem that the VGA BIOS thought the card was the EGA model and set it up like EGA. Looking back I think it was doing a read/write test with a VGA register that is 7 bits wide on some cards and 8 bits wide on another, and VGA register emulation didn't emulate it as it expected.

**Comment by joncampbell123** [comment]
S3 ViRGE emulation is complete as far as video modes and 2D acceleration is concerned as well as the YUV (YUYV) hardware overlays.

3D acceleration however, is not implemented.

**Comment by joncampbell123** [comment]
If you're asking about S3 emulation beyond ViRGE, no, it doesn't have that yet.

There's also some other specific version of S3 that I am unable to get working with Windows drivers yet.
